airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -13,6 +16,49 @@ const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
   return streamNode.config?.cursorField || [];
 };
 
+const clearBreakingFieldChanges = (nodeStream: SyncSchemaStream, breakingChangesByStream: StreamTransform[]) => {
+  if (!breakingChangesByStream?.length || !nodeStream.config) {
+    return nodeStream;
+  }
+
+  let clearPrimaryKey = false;
+  let clearCursorField = false;
+
+  for (const streamTransformation of breakingChangesByStream) {
+    // get all of the removed field paths for this transformation
+    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
+
+    if (!removedFieldPaths?.length) {
+      return nodeStream;
```

Comment:
It doesn't seem like this state is possible, but `continue;` is likely the better choice.
```suggestion
      continue;
```
Maybe even add a datadog event.

Reply:
Agree.  Returning here prevents any additional checks on the stream.  It _shouldn't_ be a code path that is ever hit, but this is safer!